Library/Homebrew/api.rb

```diff
@@ -41,6 +41,11 @@ def self.fetch_json_api_file(endpoint, target:)
       url = "#{Homebrew::EnvConfig.api_domain}/#{endpoint}"
       default_url = "#{HOMEBREW_API_DEFAULT_DOMAIN}/#{endpoint}"
 
+      if Homebrew.running_as_root_but_not_owned_by_root? &&
+         (!target.exist? || target.empty?)
+        odie "Need to download #{url} but cannot as root! Try again without `sudo`."
```

Comment:
`brew services` changes its behaviour based on `sudo` so probably best to direct them to a different command.

```suggestion
        odie "Need to download #{url} but cannot as root! Run `brew update` without `sudo` first."
```

Reply:
@Bo98 Yeh, good call, your message is much better.